### PR TITLE
Directory Service Auth Provider Resource Implementation

### PR DESCRIPTION
**Pull request by sapana05**
##### SUMMARY
Directory Service Auth Provider Resource Implementation

##### ISSUE TYPE
- Feature Pull Request

#### Output from acceptance testing:
```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Comment by sapana05** [comment]
image
image
image

File: redfish/provider/resource_redfish_directory_service_auth_provider.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+	if err != nil {
+		diags.AddError("Error fetching Account Service", err.Error())
+		return diags
+	}
+
+	if diags = parseActiveDirectoryIntoState(ctx, accountService, service, state); diags.HasError() {
+		diags.AddError("ActiveDir state null", "ActiveDirectory state null")
+		return diags
+	}
+	// if !oldLDAPState.IsNull() && !oldLDAPState.IsUnknown() {
```
Comment (taohe1012) [inline redfish/provider/resource_redfish_directory_service_auth_provider.go:317]: this line could be removed?

File: redfish/provider/resource_redfish_directory_service_auth_provider.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+	if err != nil {
+		diags.AddError("There was an error while creating/ updating Active Directory",
+			"There was an error while creating/ updating Active Directory "+err.Error())
+		return diags
+	}
+	if response != nil {
+		body, err := io.ReadAll(response.Body)
+		if err != nil {
+			diags.AddError("error reading response body", "error "+string(body))
+			// diags.AddError("error reading response body", )
```
Comment (taohe1012) [inline redfish/provider/resource_redfish_directory_service_auth_provider.go:367]: remove this line

File: redfish/provider/resource_redfish_directory_service_auth_provider.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+		}
+	}
+
+	defer response.Body.Close()
+	var idracAttributesPlan models.DellIdracAttributes
+	idracAttributesPlan.Attributes = plan.ActiveDirectoryAttributes
+	idracAttributesPlan.ID = plan.ID
+	idracAttributesPlan.RedfishServer = plan.RedfishServer
+	diags = updateRedfishDellIdracAttributes(ctx, service, &idracAttributesPlan)
+	// resp.Diagnostics.Append(diags...)
```
Comment (taohe1012) [inline redfish/provider/resource_redfish_directory_service_auth_provider.go:391]: this line

File: redfish/provider/resource_redfish_directory_service_auth_provider.go [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+		// diags.AddError("Idrac update fail", "Idrac update fail due to invalid Active Directory configuration")
+		return diags
+	}
+	plan.ActiveDirectoryAttributes = idracAttributesPlan.Attributes
+	return diags
+}
+
+// nolint: revive
+func updateLDAP(ctx context.Context, serviceURI string, service *gofish.Service, plan *models.DirectoryServiceAuthProviderResource) (diags diag.Diagnostics) {
+	// var diags diag.Diagnostic
```
Comment (taohe1012) [inline redfish/provider/resource_redfish_directory_service_auth_provider.go:402]: this line

**Review by taohe1012 — COMMENTED** [review]
hi @sapana05, seems some useless line are not deleted.

**Review by taohe1012 — APPROVED** [review]

**Pull request merged**